File: csharp/src/Core.Abstractions/Models/TransactionRequest.cs
using Train.Solver.Core.Abstractions.Entities;

namespace Train.Solver.Core.Abstractions.Models;

public class TransactionRequest : BaseRequest
{
    public required string PrepareArgs { get; set; } = null!;

    public required TransactionType Type { get; set; }

    public required NetworkType NetworkType { get; set; }

    public required string FromAddress { get; set; } = null!;

    public required string SwapId { get; set; }
}

File: js/src/CoreAbstraction/Models/TransacitonModels/TransactionRequest.ts
import { NetworkType } from "../../../Data/Entities/Networks";
import { BaseRequest } from "../BaseRequest";
import { TransactionType } from "./TransactionType";

export interface TransactionRequest extends BaseRequest {
    PrepareArgs: string;
    TransactionType: TransactionType;
    NetworkType: NetworkType;
    FromAddress: string;
    SwapId: string;
}